TheMET/TheMET/LoadingPlaceholderView/LoadingPlaceholderView.swift

```diff
@@ -2,57 +2,59 @@
 //  LoadingPlaceholderView.swift
 //  TheMET
 //
-//  Created by Анна Ситникова on 18/04/2023.
+//  Created by Анна Ситникова on 08/08/2023.
 //
 
 import Foundation
 import UIKit
 
 class LoadingPlaceholderView: UIView {
     
-    static let xibFileName = "LoadingPlaceholderView"
+    private let textLabel: UILabel = UILabel()
+    private let animationView: ArtsLoadingIndicatorView = ArtsLoadingIndicatorView()
     
-    var onButtonTap: () -> Void = {}
-    
-    @IBOutlet var textLabel: UILabel!
-    @IBOutlet var imageView: UIImageView!
-    @IBOutlet var button: UIButton!
-    
-    static func constructView(configuration: LoadingPlaceholderConfiguration) -> LoadingPlaceholderView {
-        let nib = UINib(nibName: LoadingPlaceholderView.xibFileName, bundle: nil)
-        let view = nib.instantiate(withOwner: nil).first as! LoadingPlaceholderView
+    static func construstView(configuration: LoadingPlaceholderConfiguration) -> LoadingPlaceholderView {
+        let view = LoadingPlaceholderView()
         view.setup(configuration: configuration)
         return view
     }
     
     private func setup(configuration: LoadingPlaceholderConfiguration) {
-        self.imageView.image = configuration.image
+        self.textLabel.numberOfLines = 0
+        self.textLabel.textAlignment = .center
         self.textLabel.apply(font: NSLocalizedString("serif_font", comment: ""), color: UIColor(named: "plum"), fontSize: 18, title: configuration.text)
-        if let buttonTitle = configuration.buttonTitle {
-            self.button.apply(radius: 30, backgroundColor: UIColor(named: "blueberry"), fontColor: UIColor(named: "pear"), font: NSLocalizedString("san_serif_font", comment: ""), fontSize: 20, buttonTitle: buttonTitle, image: nil)
-            self.button.addTarget(self, action: #selector(buttonDidTap), for: .touchUpInside)
-        } else {
-            self.button.isHidden = true
-        }
     }
     
-    @objc
-    private func buttonDidTap(){
-        self.onButtonTap()
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.animationView.translatesAutoresizingMaskIntoConstraints = false
+        self.textLabel.translatesAutoresizingMaskIntoConstraints = false
+        self.addSubview(self.animationView)
+        self.addSubview(self.textLabel)
+        NSLayoutConstraint.activate([
+            self.animationView.heightAnchor.constraint(equalTo: self.heightAnchor, multiplier: 0.6),
+            self.animationView.widthAnchor.constraint(equalTo: self.widthAnchor, multiplier: 0.5),
+            self.animationView.centerXAnchor.constraint(equalTo: self.centerXAnchor),
+            self.animationView.topAnchor.constraint(equalTo: self.topAnchor, constant: 30),
```

Comment:
Not really flexible solution. On all the screens the animation-view if going to be attached to the top by 30px.
It is better to put animation-view and label to container and center the container in the view

Reply:
BTW try to use UIStackView as a container.